packages/js/product-editor/src/hooks/use-conditional-execution.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { RefObject } from 'react';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useVisibilityObserver } from './use-visibility-observer';
+
+type useConditionalExecutionProps = {
+	elementOrSelector?: RefObject< HTMLInputElement > | string;
+	onVisible?: () => void;
+	onHidden?: () => void;
+	isMemorized?: boolean;
+};
+
+export function useConditionalExecution( {
+	elementOrSelector,
+	onVisible,
+	onHidden,
+	isMemorized = false,
```

Comment:
We use the `isMemorized` to avoid adding the `onVisible` or `onHidden` functions as a dependencies. Adding a non-memorized function as a dependency for the `useEffect` would trigger an infinite loop.

Reply:
Not sure about this. Check my comments below.